lib/mayaUsd/fileio/primReader.h

```diff
@@ -31,6 +31,24 @@ class UsdMayaPrimReader
     UsdMayaPrimReader(const UsdMayaPrimReaderArgs&);
     virtual ~UsdMayaPrimReader() {};
 
+    /// The level of support a reader can offer for a given context
+    ///
+    /// A basic reader that gives correct results across most contexts should
+    /// report `Fallback`, while a specialized reader that really shines in a
+    /// given context should report `Supported` when the context is right and
+    /// `Unsupported` if the context is not as expected.
+    enum class ContextSupport
+    {
+        Supported,
+        Fallback,
+        Unsupported
+    };
+
+    /// This static function is expected for all shader readers and allows
+    /// declaring how well this class can support the current context:
+    MAYAUSD_CORE_PUBLIC
+    static ContextSupport CanImport(const UsdMayaJobImportArgs& importArgs);
```

Comment:
That feature would be harder to impement and it may have a negative performance impact during the import, so I would wait with implementing it until we have a specific customer request.

Reply:
Actually, I just double-checked and we call UsdMayaPrimReaderRegistry::FindOrFallback for every single prim we exprort, so implementing this feature will not be that hard as I thought. And since we call FindOrFallback on every prim (although we don't have to now), the performance impact is already there.

Reply:
@samuelliu-adsk - it would not solve the issue as I am not suggesting to replace the default Maya reader but to enable the reader registry to make a better discrimination from the available readers for a prim type. The same Prim type could contain or not vendor specific attributes that would be of interest for a given reader. If the prim attributes can be inspected at the CanImport call, it would enable the reader to report it supports the specialized prim or not, otherwise falling back to the default Maya reader. That is my suggestion. I am not against the JobContext idea, but I think that using the CanImport facility is easier on users for now.

The 3ds Max prim writers are working in a similar fashion and it brings flexibility.

Reply:
Hi @calligrapheric. Good point. For this PR I think we can focus on CanImport for now.

> a given LightPrim having attributes specializing the import (using a specific reader) but otherwise, the user would default back to the Maya Light importer
After some thoughts, I think it's a good design but I also doubt inspecting the prim on our side will do much help... 
1. What if the customer wants to override the default prim reader regardless it contains special attributes or not? In that case we will be forcing it back to default reader..
2. Shouldn't the customized prim reader be able to handle non-special attributes already? If not they can't read prims properly..

Let me know if I'm wrong! I would love to hear more suggestions from you!

Reply:
The `CanImport` static method only receives the import args. I propose we add the UsdPrim the PrimReader is asked to act on. To answer your points:

1. In my proposition, the PrimReader has the _option_ of inspecting the UsdPrim. To override the default reader, it simply needs to reply it is a supported reader regardless of the UsdPrim content, as in your PR. If two plugins provide a PrimReader supporting the same prim type, then we would need Neil's JobContext for the user to select the one to keep.
2. In the actual state, the PrimReader `Read` method can handle all attributes (standard or vendor specific). That is not the question. But the `CanImport` static method has no clue on which UsdPrim (besides its type) it is replying.